phac-nml/irida-next
# Sample files updated column

## What does this PR do and why?
_Describe in detail what your merge request does and why._

This PR adds a new column to the samples table `attachments_updated_at` which gets updated whenever files are added or removed to/from a sample.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

![image](https://github.com/phac-nml/irida-next/assets/25466211/60b8239c-b826-4fce-9d9e-405ad736b7cf)

**No attachments**

![image](https://github.com/phac-nml/irida-next/assets/25466211/8c4f82be-1d02-4055-8876-25f4ac7c3cfe)

**Uploaded attachments  - samples table at project level**

![image](https://github.com/phac-nml/irida-next/assets/25466211/3aec2807-4f00-4c5d-9cb1-799db5cbcacc)

**Destroyed attachments - samples table at project level**

![image](https://github.com/phac-nml/irida-next/assets/25466211/e3efa1f9-1eca-4c0c-8b2f-af53848090cd)

**Samples table at group level**

![image](https://github.com/phac-nml/irida-next/assets/25466211/7c608028-c035-4eef-a4fb-6d4c0bac4f88)

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

1. Switch to the main branch
2. Reset your database `bin/rails db:drop db:create db:migrate db:seed`
3. Switch to this branch
4. Run `bin/rails db:migrate`
5. Open `rails console` in the terminal
6. Run the query `Sample.first` and make note of the new `attachments_updated_at` column and its timestamp
7. Run the query `Sample.last`. The timestamps for the `attachments_updated_at` columns should be different
8. Login to IRIDA Next 
9. Create a project under a group, add a sample, upload some files to the sample
10. View the project samples by clicking the `Samples` link in the left sidebar
11. Table should display timestamp for `Files Last Updated`
12. View the samples for the group (the group the project was created under in step 9)
13. You should see the table has a column for `Files Last Updated`

**Note:** The sample `updated_at` column is updated alongside the `attachments_updated_at` column when attachments are uploaded/removed for a sample. This is the behaviour of the `touch` option for `belongs_to`. 

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.
